apps/opencs/view/render/terrainstorage.cpp

```diff
@@ -33,10 +41,118 @@ namespace CSVRender
         return &mData.getLandTextures().getRecord(row).get();
     }
 
+    void TerrainStorage::setAlteredHeight(int inCellX, int inCellY, float height)
+    {
+        mAlteredHeight[inCellY*ESM::Land::LAND_SIZE + inCellX] = height - fmod(height, 8); //Limit to divisible by 8 to avoid cell seam breakage
+    }
+
+    void TerrainStorage::resetHeights()
+    {
+        std::fill(std::begin(mAlteredHeight), std::end(mAlteredHeight), 0);
+    }
+
+    float TerrainStorage::getSumOfAlteredAndTrueHeight(int cellX, int cellY, int inCellX, int inCellY)
+    {
+        float height = 0.f;
+        osg::ref_ptr<const ESMTerrain::LandObject> land = getLand (cellX, cellY);
+        if (land)
+        {
+            const ESM::Land::LandData* data = land ? land->getData(ESM::Land::DATA_VHGT) : 0;
```

Comment:
```suggestion
            const ESM::Land::LandData* data = land ? land->getData(ESM::Land::DATA_VHGT) : nullptr;
```

Reply:
Good point! This method is also used in here btw: https://github.com/OpenMW/openmw/blob/5d5ec1cf639cabb3aef8608adbcd2b6f592195ff/components/esmterrain/storage.cpp#L77

Reply:
committed in fdcc7fa , but maybe the one in storage.cpp should be fixed in another PR...